queries.yaml

```diff
@@ -794,6 +794,41 @@ post_process:
       where: >-
         kind == 'lake'
 
+  # remove names per kind for a given zoom
+  # before generating any label placements
+  - fn: vectordatasource.transform.drop_properties
+    params:
+      source_layer: water
+      start_zoom: 0
+      end_zoom: 16
+      # short-hand for "name" property in the list below means all name-like
+      # properties.
+      all_name_variants: true
+      properties:
+        - name
+        - old_name
+      where: >-
+        (kind == 'sea' and zoom < 4) or
+        (kind == 'bay' and zoom < 5) or
+        (kind == 'fjord' and zoom < 5) or
+        (kind == 'strait' and zoom < 5) or
+        (kind == 'lake' and zoom < 5) or
+        (kind == 'playa' and zoom < 6) or
+        (kind == 'reef' and zoom < 6) or
+        (kind == 'river' and zoom <  12) or
+        (kind == 'riverbank' and zoom <  12) or
+        (kind == 'water' and zoom <  12) or
+        (kind == 'canal' and zoom <  13) or
+        (kind == 'basin' and zoom <  13) or
+        (kind == 'dock' and zoom <  13) or
```

Comment:
I expanded the list of kinds in this section based on QA where I saw low zoom `basin`, `dock` and `straight` points around London and the English Channel. Now it has every water kind.